### PR TITLE
Provide a standard way to import a database to drupal

### PR DESCRIPTION
**Pull request by kierenevans**
Builds upon #174 to provide standard ways of importing a database dump in line with the other framework images.

This includes the means to fetch that database dump via SSH if certain environment variables are configured and the `do_drupal_sync_database` function is called from a custom version of `do_setup` or `do_drupal_install`.

The PR also includes the ability to run drupal's composer plugin if required, via `do_drupal_composer_install`.

### CODE REVIEW

File: drupal/usr/local/share/drupal/drupal_functions.sh [hunk 1]
```diff
@@ -90,20 +90,104 @@ do_drupal_install() {
...
+
+  echo 'Copy the database from the remote server to the container'
+  as_build "scp -i '/home/build/.ssh/${DRUPAL_SYNC_SSH_KEY_NAME}' -P '${DRUPAL_SYNC_SSH_SERVER_PORT}' '${DRUPAL_SYNC_SSH_USERNAME}@${DRUPAL_SYNC_SSH_SERVER_HOST}:${DATABASE_BACKUP_REMOTE_PATH}' '${DATABASE_ARCHIVE_PATH}'"
+}
+
+#####
+# Restore a database if required.
+# Not triggered by default, please call from a function in your plan.sh to use.
+#####
+do_drupal_database_install() {
```
Comment (rj-ikos) [inline drupal/usr/local/share/drupal/drupal_functions.sh]: Please can we change the function name to do_drupal_database_restore

File: drupal/usr/local/share/drupal/drupal_functions.sh [hunk 2]
```diff
@@ -90,20 +90,104 @@ do_drupal_install() {
...
+    fi
+  fi
+  set -x
+}
+
+do_drupal_database_sanitise() {
+  as_code_owner "drush ${DRUPAL_DRUSH_ALIAS} sql-sanitize --yes -r ${WEB_DIRECTORY}"
+}
+
+do_drupal_composer_install() {
```
Comment (rj-ikos) [inline drupal/usr/local/share/drupal/drupal_functions.sh]: This function would be project specific. It doesn't need to be in the container.

File: drupal/usr/local/share/drupal/drupal_functions.sh [hunk 3]
```diff
@@ -90,20 +90,104 @@ do_drupal_install() {
   # Drop the database if we need to force the install every time.
   if [ "${FORCE_DATABASE_DROP}" == 'true' ]; then
     echo 'Dropping the Drupal DB if it exists'
-    as_code_owner "drush sql-drop -y -r ${WEB_DIRECTORY}"
+    as_code_owner "drush ${DRUPAL_DRUSH_ALIAS} sql-drop -y -r ${WEB_DIRECTORY}"
```
Comment (rj-ikos) [inline drupal/usr/local/share/drupal/drupal_functions.sh]: Change to `drush sql-drop` to eliminate the risk of accidentally destroying data in the wrong environment.

File: drupal/usr/local/share/drupal/drupal_functions.sh [hunk 4]
```diff
@@ -90,20 +90,104 @@ do_drupal_install() {
...
     echo 'Installing Drupal'
-
+    set +x
     INSTALL_OPTS="${DRUPAL_INSTALL_PROFILE} --account-name=\"${DRUPAL_ADMIN_USERNAME}\"  --account-pass=\"${DRUPAL_ADMIN_PASSWORD}\" install_configure_form.update_status_module='array(FALSE,FALSE)'"
     # We should make sure this is writeable, but only do it directly before an
     # install. Drupal will lock it back down on install completion.
     chmod -R ug+rw,o-w "${WEB_DIRECTORY}/sites/default/files"
     chmod go+w "${WEB_DIRECTORY}/sites/default/settings.php"
-    as_code_owner "drush site-install ${INSTALL_OPTS} -y -r ${WEB_DIRECTORY}"
+    as_code_owner "drush ${DRUPAL_DRUSH_ALIAS} site-install ${INSTALL_OPTS} -y -r ${WEB_DIRECTORY}"
```
Comment (rj-ikos) [inline drupal/usr/local/share/drupal/drupal_functions.sh]: As above, do not use alias here as there is a risk of destroying data elsewhere.

File: drupal/usr/local/share/env/30-framework [hunk 5]
```diff
@@ -5,10 +5,41 @@ export WEB_DIRECTORY=${WEB_DIRECTORY:-docroot}
 export DRUPAL_DATABASE_NAME=${DRUPAL_DATABASE_NAME:-drupaldb}
```
Comment (rj-ikos) [inline drupal/usr/local/share/env/30-framework]: Please update the `readme` with these additional functions and variables.

**Comment by sroze** [comment]
@kierenevans should we ask @rj-ikos & @andytson-inviqa to review again?

**Comment by sroze** [comment]
@rj-ikos, @andytson-inviqa: could you update your review on this PR?

**Comment by rj-ikos** [comment]
Happy for this to be merged now. 
Will leave this comment for approval and allow Kieren to merge when ready.

**Pull request merged**